File: day17/index2.js
const STEP = 316;
const MAX = 5e7;

let i = 0;
let zeroIndex = 0;
let ans = null;

for (let v = 1; v <= MAX; v++) {
  i = (i + STEP) % v;
  if (i < zeroIndex) {
    zeroIndex++;
  } else if (i === zeroIndex) {
    ans = v;
  }
  i += 1;
}

console.log(ans);

File: clojure/day17/index2.clj
(def STEP 316)
(def MAX 5e7)

(def res (loop [
  i 0
  zero-ind 0
  ans nil
  v 1
] (if (> v MAX) ans
    (let [
      i' (mod (+ i STEP) v)
      zero-ind' (if (< i' zero-ind) (inc zero-ind) zero-ind)
      ans' (if (== i' zero-ind) v ans)
    ] (recur (inc i') zero-ind' ans' (inc v))))))

(println res)
